client/state/data-layer/wpcom/sites/media/index.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * Internal dependencies
+ */
+import { MEDIA_REQUEST } from 'state/action-types';
+import { receiveMedia, requestingMedia } from 'state/media/actions';
+import wpcom from 'lib/wp';
+
+/**
+ * Issues an API request to fetch media for a site and query.
+ *
+ * @param  {Object}         store  Redux store
+ * @param  {Object}         action Action object
+ * @return {XMLHttpRequest}        XMLHttpRequest
+ */
+export function requestMedia( { dispatch }, { siteId, query } ) {
+	dispatch( requestingMedia( siteId, query ) );
+
+	return wpcom.site( siteId ).mediaList( query, function( error, data ) {
```

Comment:
At the very least I might suggest the more concise ES6 arrow function syntax for the anonymous function, though further I wonder if we should treat the return value as a promise. This would be difficult to test as-is, for example, since nock stubbing doesn't complete predictably in the current or next tick.